python/cudf/cudf/dataframe/series.py

```diff
@@ -1574,7 +1574,10 @@ def one_hot_encoding(self, cats, dtype="float64"):
         A sequence of new series for each category.  Its length is determined
         by the length of ``cats``.
         """
-
+        if isinstance(cats, Series):
```

Comment:
This needs to be generalized more, either need to hit `Index` as well, or do something like `if hasattr(cats, "to_pandas")`

Reply:
Made it generic.